docs/api-markdown-documenter/front-matter.js

```diff
@@ -0,0 +1,115 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+const {
+	createDocumentWriter,
+    getLinkForApiItem,
+    getUnscopedPackageName,
+	renderNodeAsMarkdown,
+    transformDocNode,
+	getMarkdownRenderContextWithDefaults,
+} = require("@fluid-tools/api-markdown-documenter");
+const { ApiItemKind } = require("@microsoft/api-extractor-model");
+const os = require("os");
+
+const generatedContentNotice = "[//]: # (Do not edit this file. It is automatically generated by @fluidtools/api-markdown-documenter.)";
+
+/**
+ * Creates Hugo front-matter for the given API item.
+ * This will be appended to the top of the generated API documents.
+ *
+ * @param {ApiItem} apiItem - The root API item of the document being rendered.
+ * @param {MarkdownDocumenterConfiguration} config - See
+ * {@link @fluid-tools/api-markdown-documenter#MarkdownDocumenterConfiguration}.
+ *
+ * @returns The JSON-formatted Hugo front-matter as a `string`.
+ */
+function createHugoFrontMatter(apiItem, config, customRenderers) {
+    function extractSummary() {
+		const summaryParagraph = transformDocNode(
+			apiItem.tsdocComment.summarySection,
+			apiItem,
+			config);
+		
+		if (!summaryParagraph) {
+			return "";
+		}
+		
+		const documentWriter = createDocumentWriter();
+		renderNodeAsMarkdown(summaryParagraph, documentWriter, getMarkdownRenderContextWithDefaults({
+			renderers: customRenderers
+		}));
+		return documentWriter.getText().replace(/"/g, "'").trim();
+    }
+
+    const frontMatter = {};
+    frontMatter.title = apiItem.displayName.replace(/"/g, '').replace(/!/g, '');
```

Comment:
FWIW, if you need to parse markdown with front-matter, I highly recommend [gray-matter](https://www.npmjs.com/package/gray-matter).

Reply:
Nice! Yeah, I'll keep that in mind. For now, I don't think we're parsing any existing frontmatter, just writing it.

Reply:
Along the same lines as serializing AST nodes... I think if you create a gray-matter object (which is just the frontmatter as an object with another property that is the markdown content), you can then just serialize it to a file without messing with strings.